lib/diane/main.go

```diff
@@ -223,7 +223,13 @@ func (parser *dianeParser) ParseLines(parsedLineChan chan marshal.ParsedLineResu
 // parseDianeRow construit un objet Diane à partir d'une ligne de valeurs récupérée depuis un fichier
 func parseDianeRow(idx marshal.ColMapping, row []string) (diane Diane) {
 
-	if i, err := strconv.Atoi(row[idx["Annee"]]); err == nil {
+	idxRow := idx.IndexRow(row)
+
+	// Cas où on cherche à récupérer la valeur d'une colonne qui n'existe pas:
+	// idxRow.GetVal("Année") // => log.Fatal: "Column not found in ColMapping: Année"
+	// ... alors que row[idx["Année"]] aurait échoué silencieusement, en retournant un indice incorrect (0)
+
+	if i, err := strconv.Atoi(idxRow.GetVal("Annee")); err == nil {
```

Comment:
on pourrait meme aller plus loin dans le sens que Pierre avait commencé à explorer: ajouter des méthodes d'affectation par type dans `IndexedRow`, ex:

```go
func (indexedRow IndexedRow) GetString(colName string) (string, error) { ... }
func (indexedRow IndexedRow) GetInteger(colName string) (int, error) { ... }
func (indexedRow IndexedRow) GetFloat(colName string) (float64, error) { ... }
```

... de manière à pouvoir alléger un peu le code des parseurs:

```suggestion
	if annee, err := idxRow.GetInteger("Annee"); err == nil {
```

Reply:
... ou aller encore un poil plus loin:

```go
  idxRow.AssignInteger("Annee", &diane.Annee); // va populer diane.Annee seulement si une valeur de type int est bien trouvée pour la colonne "Annee"
```

Reply:
> on pourrait meme aller plus loin dans le sens que Pierre avait commencé à explorer: ajouter des méthodes d'affectation par type dans `IndexedRow`, ex:
> 
> ```go
> func (indexedRow IndexedRow) GetString(colName string) (string, error) { ... }
> func (indexedRow IndexedRow) GetInteger(colName string) (int, error) { ... }
> func (indexedRow IndexedRow) GetFloat(colName string) (float64, error) { ... }
> ```
> 
> ... de manière à pouvoir alléger un peu le code des parseurs:
> 
> Qu'en dis tu ?

je trouve que c'est cette solution qui est la plus simple, l'autre solution allant un peu plus loin demanderait de charger les structs avec des accesseurs pour aller assigner les valeurs, ça me parait un poil plus lourd à mettre en œuvre.

Reply:
merci d'avoir partagé ton avis ! ça me va ! 👌